modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceCountryHelperImpl.java

```diff
@@ -0,0 +1,156 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.util;
+
+import com.liferay.commerce.inventory.model.CommerceInventoryWarehouseTable;
+import com.liferay.commerce.product.model.CommerceChannelRelTable;
+import com.liferay.commerce.util.CommerceCountryHelper;
+import com.liferay.petra.sql.dsl.DSLQueryFactoryUtil;
+import com.liferay.petra.sql.dsl.expression.Predicate;
+import com.liferay.petra.sql.dsl.query.FromStep;
+import com.liferay.petra.sql.dsl.query.GroupByStep;
+import com.liferay.petra.sql.dsl.query.JoinStep;
+import com.liferay.portal.kernel.model.Country;
+import com.liferay.portal.kernel.model.CountryTable;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.CountryLocalService;
+import com.liferay.portal.kernel.util.OrderByComparatorFactoryUtil;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(
+	enabled = false, immediate = true, service = CommerceCountryHelper.class
+)
+public class CommerceCountryHelperImpl implements CommerceCountryHelper {
+
+	public List<Country> getBillingCountriesByChannelId(
+		long channelId, int start, int end) {
+
+		return _countryLocalService.dslQuery(
+			_getGroupByStep(
+				DSLQueryFactoryUtil.selectDistinct(CountryTable.INSTANCE),
+				channelId, true, false
+			).orderBy(
+				CountryTable.INSTANCE,
+				OrderByComparatorFactoryUtil.create("Country", "position", true)
+			).limit(
+				start, end
+			));
+	}
+
+	public List<Country> getShippingCountriesByChannelId(
+		long channelId, int start, int end) {
+
+		return _countryLocalService.dslQuery(
+			_getGroupByStep(
+				DSLQueryFactoryUtil.selectDistinct(CountryTable.INSTANCE),
+				channelId, false, true
+			).orderBy(
+				CountryTable.INSTANCE,
+				OrderByComparatorFactoryUtil.create("Country", "position", true)
+			).limit(
+				start, end
+			));
+	}
+
+	public List<Country> getWarehouseCountries(long companyId, boolean all) {
+		return _countryLocalService.dslQuery(
+			DSLQueryFactoryUtil.selectDistinct(
+				CountryTable.INSTANCE
+			).from(
+				CountryTable.INSTANCE
+			).innerJoinON(
+				CommerceInventoryWarehouseTable.INSTANCE,
+				CountryTable.INSTANCE.a2.eq(
+					CommerceInventoryWarehouseTable.INSTANCE.
+						countryTwoLettersISOCode)
+			).where(
+				() -> {
+					Predicate predicate =
+						CommerceInventoryWarehouseTable.INSTANCE.companyId.eq(
+							companyId);
+
+					if (all) {
```

Comment:
I think this condition needs to be flipped. In Commerce's custom query, they remove the extra predicate statement if `all` is true.

@brianchandotcom if the rest of the pull looks good, would you be willing to flip this for us?